nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/FetchHBaseRow.java

```diff
@@ -0,0 +1,407 @@
+            .name("Row Identifier")
+            .description("The identifier of the row to fetch.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor COLUMNS = new PropertyDescriptor.Builder()
+            .name("Columns")
+            .description("An optional comma-separated list of \"<colFamily>:<colQualifier>\" pairs to fetch. To return all columns " +
+                    "for a given family, leave off the qualifier such as \"<colFamily1>,<colFamily2>\".")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.createRegexMatchingValidator(COLUMNS_PATTERN))
+            .build();
+
+    static final AllowableValue DESTINATION_ATTRIBUTES = new AllowableValue("flowfile-attributes", "flowfile-attributes",
+            "Adds each cell as a FlowFile attribute where they key is col-family:col-qualifier and the value is the cell's value.");
+    static final AllowableValue DESTINATION_CONTENT = new AllowableValue("flowfile-content", "flowfile-content",
+            "Overwrites the FlowFile content with a JSON document representing the row that was fetched. " +
+                    "The format of the JSON document is determined by the JSON Format property.");
+
+    static final PropertyDescriptor DESTINATION = new PropertyDescriptor.Builder()
+            .name("Destination")
+            .description("Indicates whether the row fetched from HBase is written to FlowFile content or FlowFile Attributes.")
+            .required(true)
+            .allowableValues(DESTINATION_ATTRIBUTES, DESTINATION_CONTENT)
+            .defaultValue(DESTINATION_ATTRIBUTES.getValue())
+            .build();
+
+    static final AllowableValue VALUE_ENCODING_STRING = new AllowableValue("string", "string", "Creates a String using the bytes of the cell value and the given Character Set.");
+    static final AllowableValue VALUE_ENCODING_BASE64 = new AllowableValue("base64", "base64", "Creates a Base64 encoded String of the cell value.");
+
+    static final PropertyDescriptor VALUE_ENCODING = new PropertyDescriptor.Builder()
+            .name("Value Encoding")
+            .description("Specifies how to represent the values of cells when stored in FlowFile attributes, or written to JSON.")
+            .required(true)
+            .allowableValues(VALUE_ENCODING_STRING, VALUE_ENCODING_BASE64)
+            .defaultValue(VALUE_ENCODING_STRING.getValue())
+            .build();
+
+    static final AllowableValue JSON_FORMAT_FULL_ROW = new AllowableValue("full-row", "full-row", "Creates a JSON document with the format: " +
+            "{\"row\": \"<row key>\", \"cells\": { \"<cell 1 family>:<cell 1 qualifier>\": \"<cell 1 value>\", \"<cell 2 family>:<cell 2 qualifier>\": \"<cell 2 value>\", ... }}.");
```

Comment:
I think there are arguments for flat and arguments for nested. Certainly one timestamp+value per column and having multiple columns in a row are common cases. HBase caps the size of a row at 1MB by default, so large rows would definitely bloat the responses from NiFi.

If it were me, I'd lean towards the less duplication and use the nested variety, but, honestly, I think either is fine and it's not so important that it requires a big investigation ;)